solidity/dashboard/src/services/staking-port-backer.service.js

```diff
@@ -0,0 +1,193 @@
+        filter: {
+          operator: [
+            ...operatorsAddressesSet,
+            ...granteeOperatorsSet,
+            ...managedGrantOperators,
+          ],
+        },
+      })
+    ).map((_) => _.returnValues.operator)
+  }
+
+  const filteredOwnedOperators = Array.from(operatorsAddressesSet).filter(
+    filterOutByOperator(operatorsToSkip)
+  )
+  const filteredGranteeOperators = Array.from(granteeOperatorsSet).filter(
+    filterOutByOperator(operatorsToSkip)
+  )
+  const filteredManagedGrantOperators = managedGrantOperators.filter(
+    filterOutByOperator(operatorsToSkip)
+  )
+
+  const ownedDelegations = await getDelegations(
+    filteredOwnedOperators,
+    undelegationPeriod
+  )
+
+  const granteeDelegations = await getDelegations(
+    filteredGranteeOperators,
+    undelegationPeriod,
+    true
+  )
+
+  const managedGrantsDelegations = await getDelegations(
+    filteredManagedGrantOperators,
+    undelegationPeriod,
+    true,
+    true
+  )
+
+  const delegations = [
+    ...ownedDelegations,
+    ...granteeDelegations,
+    ...managedGrantsDelegations,
+  ].sort((a, b) => sub(b.createdAt, a.createdAt))
+
+  return {
+    delegations,
+    undelegationPeriod,
+  }
+}
+
+const getDelegations = async (
+  operators,
+  undelegationPeriod,
+  isFromGrant,
+  isManagedGrant
+) => {
+  const web3 = await Web3Loaded
+  const { oldTokenStakingContract, grantContract } = await ContractsLoaded
+
+  const delegations = []
+
+  for (const operatorAddress of operators) {
+    const {
+      createdAt,
+      undelegatedAt,
+      amount,
+    } = await oldTokenStakingContract.methods
+      .getDelegationInfo(operatorAddress)
+      .call()
+
+    const beneficiary = await oldTokenStakingContract.methods
+      .beneficiaryOf(operatorAddress)
+      .call()
+    const authorizerAddress = await oldTokenStakingContract.methods
+      .authorizerOf(operatorAddress)
+      .call()
+
+    let grantId = null
+    let managedGrantContractInstance = null
+    if (isFromGrant) {
+      try {
+        const grantStakeDetails = await grantContract.methods
+          .getGrantStakeDetails(operatorAddress)
+          .call()
+        grantId = grantStakeDetails.grantId
+      } catch (error) {
+        grantId = null
+      }
+    }
+
+    if (isManagedGrant && grantId) {
+      const { grantee } = await grantContract.methods.getGrant(grantId).call()
+      managedGrantContractInstance = createManagedGrantContractInstance(
+        web3,
+        grantee
+      )
+    }
+
+    const operatorData = {
+      undelegatedAt,
+      amount,
+      beneficiary,
+      operatorAddress,
+      createdAt,
+      authorizerAddress,
+      isFromGrant,
+      grantId,
+      isManagedGrant,
+      managedGrantContractInstance,
+    }
+    const balance = web3Utils.toBN(amount)
+
+    if (!balance.isZero() && operatorData.undelegatedAt === "0") {
+      operatorData.isUndelegation = false
```

Comment:
We want to mark the delegation that is currently delegating, right? Can we rename the property to `isUndelegating` across the code? 

```suggestion
      operatorData.isUndelegating = false
```

Reply:
pls see b642685